### PR TITLE
refactor(createRoutes): simplify and DRY routes and remove comments

### PR DESCRIPTION
**Pull request by devopsmariocom**

### CODE REVIEW

File: src/routes.tsx [hunk 1]
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
...
-      },
-    },
-    {
-      path: 'faq',
-      // name: 'faq',
-      getComponent( nextState, cb ) {
-        const importModules = Promise.all( [
-          System.import( './faq/Faq' ),
-        ] );
+        importModules.catch( ( err: any ) => {
```
Comment (Hotell) [inline src/routes.tsx]: Chain

File: src/routes.tsx [hunk 2]
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
...
+          System.import( `./${nextState.location.pathname.slice(1)}/index` ),
         ] );
 
         const renderRoute = loadModule( cb );
 
-        importModules.then( ( [component] ) => {
-          renderRoute( component.About );
+        importModules.then( ( [
+          component ] ) => {
+          renderRoute( component[ Object.keys( component )[0] ] );
```
Comment (Hotell) [inline src/routes.tsx]: I don't like this

File: src/routes.tsx [hunk 3]
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
...
-    //   // },
-    // },
     {
-      path: 'about',
-      // name: 'about',
+      path: '*',
       getComponent( nextState, cb ) {
         const importModules = Promise.all( [
-          System.import( './about/About' ),
+          System.import( `./${nextState.location.pathname.slice(1)}/index` ),
```
Comment (Hotell) [inline src/routes.tsx]: Btw this Promise.all is unnecessary in this case, we are loading just one file/module. Root one

File: src/routes.tsx (hunk 2 above)
Comment (devopsmariocom) [inline src/routes.tsx]: me neither any suggestions?

File: src/routes.tsx [hunk 4]
```diff
@@ -1,85 +1,40 @@
...
 
 import { App } from './App';
 import { Home } from './home';
-import { About } from './about';
-import { Faq } from './faq';
-
-const errorLoading = ( err: any ) => {
-  console.error( 'Dynamic page loading failed', err );
-};
+import { NotFound } from './not-found/NotFound';
```
Comment (Hotell) [inline src/routes.tsx]: we are importing from barrel, not from file directly

File: src/routes.tsx [hunk 5]
```diff
@@ -1,85 +1,40 @@
...
 
 const loadModule = ( cb ) => ( componentModule ) => {
   cb( null, componentModule );
 };
 
+const getObjectFirstKey = ( object ) => Object.keys( object )[0];
```
Comment (Hotell) [inline src/routes.tsx]: no types, what is object?

File: src/routes.tsx [hunk 6]
```diff
@@ -1,85 +1,40 @@
...
+
+const removeLeadingSlash = ( routePath ) => routePath.slice( 1 );
```
Comment (Hotell) [inline src/routes.tsx]: no types, what is routePath? if string, consider `substring` `slice` is more supposed to be used with Array

File: src/routes.tsx [hunk 7]
```diff
@@ -1,85 +1,40 @@
...
+      path: '*',
       getComponent( nextState, cb ) {
-        const importModules = Promise.all( [
-          System.import( './about/About' ),
-        ] );
 
-        const renderRoute = loadModule( cb );
+        System.import( `./${ removeLeadingSlash( nextState.location.pathname ) }/index` )
+          .then( ( component ) => {
+            loadModule( cb )( component[ getObjectFirstKey( component ) ] );
```
Comment (devopsmariocom) [inline src/routes.tsx]: still it more feels like default exports make sense for page components WDYT?

see https://github.com/ngParty/React-scaffold/pull/18

File: src/routes.tsx [hunk 8]
```diff
@@ -3,15 +3,21 @@ import { hashHistory, Router, RouteConfig } from 'react-router';
 
 import { App } from './App';
 import { Home } from './home';
-import { NotFound } from './not-found/NotFound';
+import { NotFound } from './not-found';
+import StatelessComponent = React.StatelessComponent;
+import Component = React.Component;
```
Comment (Hotell) [inline src/routes.tsx]: This namespace aliasing is wrong. Pls watch what webstorm generates for your next time before committing

**Comment by devopsmariocom** [comment]
merging reason: we have to move on, any changes to current state can be and will be easily reverted

**Pull request merged**